src/base/os.lua

```diff
@@ -134,26 +134,65 @@
 	end
 
 
-
 --
--- Retrieve the current operating system ID string.
+-- Retrieve the current target operating system ID string.
 --
 
+	function os.target()
+		return _OPTIONS.os or _TARGET_OS
+	end
+
 	function os.get()
-		return _OPTIONS.os or _OS
+		premake.warnOnce("os.get", "os.get() is deprecated, use 'os.target()' or 'os.host()'.")
+		return os.target()
 	end
 
+	-- deprecate _OS
+	_G_metatable = {
+		__index = function(t, k)
+			if (k == '_OS') then
+				premake.warnOnce("_OS+get", "_OS is deprecated, use '_TARGET_OS'.")
+				return rawget(t, "_TARGET_OS")
+			else
+				return rawget(t, k)
+			end
+		end,
+
+		__newindex = function(t, k, v)
+			if (k == '_OS') then
+				premake.warnOnce("_OS+set", "_OS is deprecated, use '_TARGET_OS'.")
+				rawset(t, "_TARGET_OS", v)
+			else
+				rawset(t, k, v)
+			end
+		end
+	}
+	setmetatable(_G, _G_metatable)

```

Comment:
This method actually has quite the impact on performance... ANY access to a global, now has a string compare first.. so I wanted to keep this as simple as possible, and really only keep it in there for the next 2-3 alpha releases. I really don't want to write a 'generic' deprecated globals table...